kafka-commons/src/main/java/uk/gov/homeoffice/digital/sas/kafka/message/KafkaEventMessage.java

```diff
@@ -1,25 +1,31 @@
 package uk.gov.homeoffice.digital.sas.kafka.message;
 
+import static uk.gov.homeoffice.digital.sas.kafka.constants.Constants.SCHEMA_FORMAT;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
 import jakarta.validation.constraints.NotNull;
 import lombok.Getter;
 
 @Getter
 public class KafkaEventMessage<T> {
 
-  public static final String SCHEMA_FORMAT = "%s, %s";
-
-  private final String schema;
+  @JsonProperty("schema")
+  private String schema;
 
   @NotNull
-  private final T resource;
+  @JsonProperty("resource")
+  private T resource;
 
   @NotNull
-  private final KafkaAction action;
+  @JsonProperty("action")
+  private KafkaAction action;
 
   public KafkaEventMessage(String schemaVersion, T resource,
       KafkaAction action) {
     this.schema = String.format(SCHEMA_FORMAT, resource.getClass().getName(), schemaVersion);
     this.resource = resource;
     this.action = action;
   }
+
+  public KafkaEventMessage() {}
```

Comment:
Do we want to allow `KafkaEventMessage` to be created anyhow? 
As in wouldn't we want to only use the other constructor which makes the `schema` property for us?

Reply:
Updated to use gson which allows us the remove this no args constructor

Reply:
great, i'm glad that worked!
Could you also re-introduce `final` on the fields now that we don't have a no args constructor.

Reply:
Done